supplement/supplementary.Rmd

```diff
@@ -138,32 +138,56 @@ The .mzML files were downloaded from the MassIVE database (accession number
 MSV000087155, available at https://massive.ucsd.edu/) via
 ftp://massive.ucsd.edu/MSV000087155. 
 
-## Instantiation of the Spectra object
+We use the [BiocFileCache]() package from Bioconductor to download and cache the
+mzML files locally. To this end we first determine below the full file names of
+all mzML files of this data set.
+
+```{r}
+url <- "ftp://massive.ucsd.edu/MSV000087155/ccms_peak/New_mzMLFinal/"
+library(curl)
+file_list <- readLines(curl(url, "r"))
+ftp_files <- strsplit(file_list, " +")
+ftp_files <- vapply(
+    ftp_files, function(z) paste0(tail(z, 2), collapse = " "), character(1))
+```
 
-In the subsequent analysis, a `Spectra` object is instantiated. The operations 
-were executed within a (high-performance) computing environment (31 cores, 
-64 GB RAM pool for all cores), where the .mzML files were stored in the directory `Cherkaoui2022`.
+With the file names available we next create a *BiocFileCache* instance adding
+the files. The *BiocFileCache* will take care of downloading files that are not
+already available in the local cache thus preventing unneeded downloads.
 
-<!-- FIXME Can you make the following (i.e. all paths) more portable by using https:// and BiocCache to access the input files, from some persistent location on the web, and if needed store computed results / intermediate files in a tempdir() location? -->
+```{r, message = FALSE}
+library(BiocFileCache)
+## FIXME: update the path; every additional result should be saved in there
+cherkaoui <- BiocFileCache("/home/jo/tmp/Cherkaoui2022", ask = FALSE)
+path <- bfcrpath(cherkaoui, paste0(url, curl_escape(ftp_files)))
```

Comment:
this will download all files and store them into the cache. no need to download files any more at any later stage.